compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/PointerInputMatcher.kt

```diff
@@ -0,0 +1,142 @@
+            override val pointerType = pointerType
+            override val button = button
+        }
+
+        @ExperimentalFoundationApi
+        fun mouse(button: PointerButton): PointerInputMatcher = MousePointerInputMatcher(button)
+        @ExperimentalFoundationApi
+        fun stylus(button: PointerButton? = null): PointerInputMatcher = StylusPointerInputMatcher(button)
+
+        @ExperimentalFoundationApi
+        val stylus: PointerInputMatcher = StylusPointerInputMatcher.Companion
+        @ExperimentalFoundationApi
+        val touch: PointerInputMatcher = TouchPointerInputMatcher
+        @ExperimentalFoundationApi
+        val eraser: PointerInputMatcher = EraserPointerInputMatcher
+
+        private interface PointerInputMatcherWithButton : PointerInputMatcher {
+            val button: PointerButton?
+
+            override fun matches(event: PointerEvent): Boolean {
+                return event.changes.fastAll { it.type == pointerType } &&
+                    if (button != null) event.button == button else true
+            }
+        }
+
+        private class MousePointerInputMatcher(
+            override val button: PointerButton
+        ) : PointerInputMatcherWithButton {
+            override val pointerType = PointerType.Mouse
+        }
+
+        private class StylusPointerInputMatcher(
+            override val button: PointerButton? = null
+        ) : PointerInputMatcherWithButton {
+            override val pointerType = PointerType.Stylus
+
+            companion object : PointerInputMatcher {
+                override val pointerType = PointerType.Stylus
+            }
+        }
+
+        private object TouchPointerInputMatcher : PointerInputMatcher {
+            override val pointerType = PointerType.Touch
+        }
+
+        private object EraserPointerInputMatcher : PointerInputMatcher {
+            override val pointerType = PointerType.Eraser
+        }
+
+        private class CombinedPointerInputMatcher(val sources: MutableList<PointerInputMatcher>) : PointerInputMatcher {
+            override val pointerType = PointerType.Unknown
+
+            override fun matches(event: PointerEvent): Boolean {
+                return sources.any { it.matches(event) }
+            }
+        }
+
+        /**
+         * The Default [PointerInputMatcher] which covers the most common cases of pointer inputs.
+         * [DefaultPointerInputMatcher] will match [PointerEvent]s, which match at least one of the following conditions:
+         * - [PointerType] is [PointerType.Mouse] and [PointerEvent.button] is [PointerButton.Primary]
+         * - [PointerType] is [PointerType.Touch]
+         * - [PointerType] is [PointerType.Stylus], regardless of any buttons pressed
+         * - [PointerType] is [PointerType.Eraser]
+         */
+        @ExperimentalFoundationApi
+        val DefaultPointerInputMatcher = mouse(PointerButton.Primary) + touch + stylus + eraser
```

Comment:
What if we name it as Primary? PointerMatcher.Primary

Reply:
It was my mistake! The intention was to rename it to `Primary`